File: 0165-compare-version-numbers/0165-compare-version-numbers.cpp
class Solution {
public:
    int compareVersion(string version1, string version2) {
        vector<int> v1;
        string temp = "";
        for (auto ch : version1) {
            if (ch == '.') {
                v1.push_back(stoi(temp));
                temp = "";
            } else {
                temp += ch;
            }
        }
        v1.push_back(stoi(temp));
        temp = "";

        vector<int> v2;
        for (auto ch : version2) {
            if (ch == '.') {
                v2.push_back(stoi(temp));
                temp = "";
            } else {
                temp += ch;
            }
        }
        v2.push_back(stoi(temp));
        temp = "";

        // for (int i = 0; i < v1.size(); i++) {
        //     cout << v1[i] << "\t";
        // }
        // cout << endl;
        // for (int i = 0; i < v2.size(); i++) {
        //     cout << v2[i] << "\t";
        // }
        // cout << endl;


        int i = 0;
        for ( ; i < v1.size() && i < v2.size(); i++) {
            if (v1[i] < v2[i]) {
                return -1;
            } else if (v1[i] > v2[i]) {
                return 1;
            }
        }

        for ( ; i < v1.size(); i++) {
            if (v1[i] != 0) {
                return 1;
            }
        }

        for ( ; i < v2.size(); i++) {
            if (v2[i] != 0) {
                return -1;
            }
        }

        return 0;
    }
};

File: 0165-compare-version-numbers/0165-compare-version-numbers.java
class Solution {
    public int compareVersion(String version1, String version2) {
        String[] splitVersion1 = version1.split("\\.");
        String[] splitVersion2 = version2.split("\\.");

        int i = 0, j = 0;
        for ( ; i < splitVersion1.length && j < splitVersion2.length; i++, j++) {
            int v1 = Integer.parseInt(splitVersion1[i]);
            int v2 = Integer.parseInt(splitVersion2[j]);

            if (v1 < v2) {
                return -1;
            } else if (v1 > v2) {
                return 1;
            }
        }

        while (i < splitVersion1.length) {
            int v1 = Integer.parseInt(splitVersion1[i]);
            int v2 = 0;

            if (v1 < v2) {
                return -1;
            } else if (v1 > v2) {
                return 1;
            }

            i++; 
        }

        
        while (j < splitVersion2.length) {
            int v1 = 0;
            int v2 = Integer.parseInt(splitVersion2[j]);

            if (v1 < v2) {
                return -1;
            } else if (v1 > v2) {
                return 1;
            }

            j++;
        }

        return 0;        
    }
}